lib/scholar/metrics/distance.ex

```diff
@@ -240,25 +368,53 @@ defmodule Scholar.Metrics.Distance do
       iex> y = Nx.tensor([1, 2, 3])
       iex> Scholar.Metrics.Distance.cosine(x, y)
       ** (ArgumentError) expected input shapes to be equal, got {2} != {3}
+
+      iex> x = Nx.tensor([[1, 2, 3], [0, 0, 0], [5, 2, 4]])
+      iex> y = Nx.tensor([[1, 5, 2], [2, 4, 1], [0, 0, 0]])
+      iex> Scholar.Metrics.Distance.cosine(x, y, axes: [1])
+      #Nx.Tensor<
+        f32[3][3]
+        [
+          [0.1704850196838379, 0.2418246865272522, 1.0],
+          [1.0, 1.0, 0.0],
+          [0.3740193247795105, 0.2843400239944458, 1.0]
+        ]
+      >
   """
   @spec cosine(Nx.t(), Nx.t()) :: Nx.t()
-  defn cosine(x, y) do
+  defn cosine(x, y, opts \\ []) do
+    cutoff = 10 * 2.220446049250313e-16
```

Comment:
What does this cutoff mean? Perhaps a comment could be useful here

Reply:
It prevents me from dividing by close to zero values in the normalization process. Similar to [link](https://github.com/scikit-learn/scikit-learn/blob/36958fb240fbe435673a9e3c52e769f01f36bec0/sklearn/preprocessing/_data.py#L88). I'll leave a comment.